.github/workflows/main.yml

```
@@ -1,6 +1,12 @@
 name: Build & Test
 
-on: push
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
```

Specifying the branches is not necessary in this case:
```suggestion
on:
  push:
  pull_request:
```

Alternatively you can use the shorthand `on: [push, pull_request]`.

---

I went with the shorthand, that 'felt more readable' to me

---

Hmmm but it looks like as-is (I didn't think to check beforehand) these events aren't exclusive. IE right now, there's a pull_request _and_ push set of checks running. Let me see what we can do about that